tests/unit/s2n_mem_usage_test.c

```diff
@@ -45,6 +53,8 @@
 /* This is roughly the current memory usage per connection, in KB */
 #ifdef __FreeBSD__
     #define MEM_PER_CONNECTION 57
+#elif defined(__OpenBSD__)
+    #define MEM_PER_CONNECTION 60
```

Comment:
I'm able to build this locally and in a github action, and the tests are passing. However, MEM_PER_CONNECTION needed to be increased for the mem_usage_test to pass in both of these environments:

```suggestion
    #define MEM_PER_CONNECTION 71
```

Reply:
Did you use https://github.com/vmactions/openbsd-builder for your GH action? I've been working with that as well (I want to submit a PR here to add OpenBSD to CI) and I noticed mem_usage_teste fails with that action. I haven't found root cause yet.

Could you share `uname -a` from your local system? Is your local system running virtualized?

Reply:
Building on a local kvm instance with only 512MB wasn't able to get this test to fail...
```
openbsd72$ vmstat    
 procs    memory       page                    disks    traps          cpu
 r   s   avm     fre  flt  re  pi  po  fr  sr wd0 cd0  int   sys   cs us sy id
 1  57   37M     92M 2051   0  48  14   0 588 395   0  257  4372  644 22  7 72
openbsd72$ uname -a
OpenBSD openbsd72.my.domain 7.2 GENERIC.MP#758 amd64
openbsd72$ ./build/bin/s2n_mem_test                                                                                                               
Running /home/doug/gitrepos/s2n-tls/tests/unit/s2n_mem_test.c ... PASSED         20 tests
```

Reply:
I did use openbsd-builder for the action. My local system is virtualized. Here's uname:
```
openbsd# uname -a
OpenBSD openbsd.amazon.com 7.2 GENERIC#728 amd64
```
The test passes on both systems with the increase to MEM_PER_CONNECTION.

Reply:
Thank you both for the uname outputs. I'm suspicious--and your data supports this--that there is something different about this test when run under the single-processor OpenBSD kernel vs the multi-proc kernel. The vmactions/openbsd-builder uses the SP kernel and that's also what @goatgoose saw a failure on. @dougch and I are testing on MP systems where the test is successful. 

I'm not yet sure how to deal with this.

Reply:
Interesting. It seems to not be related to MP vs SP. I re-installed openBSD in a VM with bsd.mp, and the test still failed for me:
```
openbsd-mp# uname -a
OpenBSD openbsd-mp.amazon.com 7.2 GENERIC.MP#758 amd64
openbsd-mp# ../../build/bin/s2n_mem_usage_test
Running /home/s2n-tls/tests/unit/s2n_mem_usage_test.c	...
Actual KB per connection: 70
FAILED test 417
```
I will keep looking into this and try to figure out why there's a difference. I'm using virtualbox as a VM. I wonder if this is somehow related? openbsd-builder also uses this VM.

Reply:
> Building on a local kvm instance with only 512MB wasn't able to get this test to fail...
> 
> ```
> openbsd72$ vmstat    
>  procs    memory       page                    disks    traps          cpu
>  r   s   avm     fre  flt  re  pi  po  fr  sr wd0 cd0  int   sys   cs us sy id
>  1  57   37M     92M 2051   0  48  14   0 588 395   0  257  4372  644 22  7 72
> openbsd72$ uname -a
> OpenBSD openbsd72.my.domain 7.2 GENERIC.MP#758 amd64
> openbsd72$ ./build/bin/s2n_mem_test                                                                                                               
> Running /home/doug/gitrepos/s2n-tls/tests/unit/s2n_mem_test.c ... PASSED         20 tests
> ```

@dougch, could you look at this again but with `s2n_mem_usage_test`? Looks like you'll have to be in `tests/unit` for it to find the PEM files.

```
~/tmp/s2n-tls/tests/unit% ../../build/bin/s2n_mem_usage_test
Running /home/joel/tmp/s2n-tls/tests/unit/s2n_mem_usage_test.c ... PASSED       1761 tests
```

Reply:
> 

Passes with 71 on my vm:

```
openbsd72$ git log -1 -q|head
commit 5b091111fb8176f6b1e241a3322989d10aae6965
Author: Joel Knight <knight.joel@gmail.com>
Date:   Tue Jan 10 21:02:02 2023 -0700

    bump MEM_PER_CONNECTION to 71 in s2n_mem_usage_test

    Per conversation with @goatgoose and @dougch, this value allows the test
    to pass on single and multi-processor test instances. The test was
    failing on SP instances with MEM_PER_CONNECTION values < 71.
openbsd72$ ../../build/bin/s2n_mem_usage_test
Running /home/doug/gitrepos/s2n-tls/tests/unit/s2n_mem_usage_test.c ... PASSED       1761 tests
```